### PR TITLE
Scrape Print Only prices when country code supplied

### PR DESCRIPTION
**Pull request by paddyo**
When a country code is supplied via the query parameter country-code, in combination with the query parameter print=1, the price for a print only offer is scraped off the subs page. This is a temporary measure until API upgrades are done. If there is any issue in retrieving the price, it will fallback to not even showing the print offer and show the trial offer instead

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -52,9 +53,11 @@
   },
   "dependencies": {
     "@financial-times/n-express": "^16.0.0",
+    "country-data": "0.0.24",
     "debug": "^2.2.0",
     "fetchres": "^1.5.1",
     "forever": "^0.15.1",
+    "isomorphic-fetch": "^2.2.1",
```
Comment (matthew-andrews) [inline package.json]: you shouldn't need to add this as a dependency…… should already be in scope……

File: server/lib/print-price-scraper.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+let fetch = require('isomorphic-fetch');
```
Comment (matthew-andrews) [inline server/lib/print-price-scraper.js]: doing this this way will mean that you lose the instrumentation we have around fetch

Comment (paddyo) [inline server/lib/print-price-scraper.js]: Where is this coming from? (in order to test).

File: server/lib/print-price-scraper.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+const url = 'https://sub.ft.com/spa_5/';
+
+module.exports = countryCode => {
+
+	if(!countryCode) {
+		return Promise.reject(new Error('No country code specified'));
+	}
+
+	return fetch(`${url}?country=${countryCode}`)
```
Comment (commuterjoy) [inline server/lib/print-price-scraper.js]: https://github.com/Financial-Times/next-metrics/blob/master/lib/metrics/services.js

^^ This is the way we instrument `fetch` - so any new host names need to be added to the file.

File: views/barriers/trial-grid.html [hunk 4]
```diff
@@ -107,7 +107,7 @@
 									<img src="{{#resize 800}}//next-geebee.ft.com/assets/print.png{{/resize}}" alt="NEWPAPER ONLY">
 								</div>
 								<div class="trial-barrier-grid__primary">
-									<p class="trial-barrier-grid__price">&pound;11 <span>per week</span></p>
+									<p class="trial-barrier-grid__price">{{{barrier.print.price}}}<span> per week</span></p>
```
Comment (commuterjoy) [inline views/barriers/trial-grid.html]: I noticed if you carry the three-letter country code over to the form it gives the right price ... https://sub.ft.com/sub/?countryCode=USA

**Comment by commuterjoy** [comment]
+1 apart from a couple of comments.

**Pull request merged**